### PR TITLE
[RJA-583][feature] Environment configurations rest/security

### PR DESCRIPTION
**Pull request by bugzmanov**

### CODE REVIEW

File: test-automation/cucumber/features/step_definitions/Configuration_steps.rb [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  config.should be_nil, "Config #{config_name} should not be seen via user #{username} account"
+end
+
+def find_config(config_name, project, username)
+  r = Genesis::Resource.new("projects/#{project_id(project)}/configs", {:username => username, :password => username})
+  r.find_by_name(config_name)
+end
+
+
+def update_config_access(project, config_id, users)
```
Comment (rsvato) [inline test-automation/cucumber/features/step_definitions/Configuration_steps.rb]: Isnt better to place all defs in helpers?

File: backend/src/main/scala/com/griddynamics/genesis/service/impl/EnvironmentAccessService.scala [hunk 2]
```diff
@@ -22,12 +22,18 @@
...
 package com.griddynamics.genesis.service.impl
 
-import com.griddynamics.genesis.api.Environment
+import com.griddynamics.genesis.api._
 import com.griddynamics.genesis.service
-import org.springframework.security.acls.domain.{BasePermission, ObjectIdentityImpl}
+import org.springframework.security.acls.domain.{GrantedAuthoritySid, PrincipalSid, BasePermission, ObjectIdentityImpl}
 import org.springframework.beans.factory.annotation.Value
+import scala._
+import scala.Predef._
```
Comment (ddurnev) [inline backend/src/main/scala/com/griddynamics/genesis/service/impl/EnvironmentAccessService.scala]: Are these imports necessary?

File: frontend/src/main/scala/com/griddynamics/genesis/rest/ConfigurationController.scala [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  @PostFilter(
+    "hasRole('ROLE_GENESIS_ADMIN')" +
+    "or hasPermission( #projectId, 'com.griddynamics.genesis.api.Project', 'administration') " +
+    "or hasPermission(filterObject, 'read')")
+  def list(@PathVariable("projectId") projectId: Int) = configRepository.list(projectId)
+
+  @RequestMapping(value = Array("{id}"), method = Array(RequestMethod.GET))
+  @ResponseBody
+  def get(@PathVariable("projectId") projectId: Int, @PathVariable("id") id: Int) =
+    configRepository.get(projectId, id).getOrElse(throw new ResourceNotFoundException("Cann't find configuration id = %d".format(id)))
```
Comment (ddurnev) [inline frontend/src/main/scala/com/griddynamics/genesis/rest/ConfigurationController.scala]: can't?

File: test-automation/cucumber/sample.properties [hunk 4]
```diff
@@ -9,3 +9,4 @@ genesis.template.repository.fs.path=templates
 
 genesis.web.admin.sha.password=aeebad4a796fcc2e15dc4c6061b45ed9b373f26adfc798ca7d2d8cc58182718e
 genesis.web.admin.username=genesis
+genesis.system.security.environment.restriction.enabled=true
```
Comment (rsvato) [inline test-automation/cucumber/sample.properties:12]: Modify also sample.properties.erb, please. This file is used to generate sample.properties for jenkins run

File: backend/src/main/scala/com/griddynamics/genesis/frontend/GenesisRestService.scala [hunk 5]
```diff
@@ -36,21 +36,30 @@ import com.griddynamics.genesis.api.Attribute
 import com.griddynamics.genesis.api.WorkflowHistory
 import com.griddynamics.genesis.api.WorkflowStep
 import com.griddynamics.genesis.api.Variable
+import com.griddynamics.genesis.api.Success
 import com.griddynamics.genesis.api.Environment
 import com.griddynamics.genesis.api.Template
 import com.griddynamics.genesis.api.StepLogEntry
 import com.griddynamics.genesis.api.VirtualMachine
 import com.griddynamics.genesis.api.BorrowedMachine
 import com.griddynamics.genesis.api.WorkflowDetails
```
Comment (ddurnev) [inline backend/src/main/scala/com/griddynamics/genesis/frontend/GenesisRestService.scala]: Maybe import com.griddynamics.genesis.api._ ?

**Pull request merged**